ui/src/app/views/workspace-list.tsx

```diff
@@ -152,20 +195,27 @@ export class WorkspaceCard extends React.Component<
       workspaceDeletionError: false});
   }
 
+  // Reloads data by calling the callback from the owning component. This
+  // currently causes the workspace-list to reload the entire list of workspaces.
+  async reloadData() {
+    await this.props.reload();
+  }
+
   render() {
-    const {wp, userEmail} = this.props;
-    const {bugReportOpen, bugReportError, confirmDeleting,
-      sharing, workspaceDeletionError} = this.state;
+    const {userEmail, wp} = this.props;
+    const {bugReportError, bugReportOpen, confirmDeleting, loadingData,
+      sharing, userRoles, workspaceDeletionError} = this.state;
     const permissionBoxColors = {'OWNER': '#4996A2', 'READER': '#8F8E8F', 'WRITER': '#92B572'};
```

Comment:
Opt: Existing error, but we should probably update these to use the colors variables if we can.

Reply:
Heh, I'll leave that as part of the broader colors cleanup. I don't want to introduce new styling bugs as part of this fix.